tiny_dnn/optimizers/optimizer.h

```diff
@@ -135,6 +136,43 @@ struct adam : public stateful_optimizer<2> {
 };
 
 /**
+ * @brief [a new optimizer (2015)]
+ * @details [see Adam: A Method for Stochastic Optimization (Algorithm 2)
+ *               http://arxiv.org/abs/1412.6980]
+ *
+ */
+struct adamax : public stateful_optimizer<2> {
+  adamax()
+    : alpha(float_t(0.002)),
+      b1(float_t(0.9)),
+      b2(float_t(0.999)),
+      b1_t(b1),
+      eps(float_t(1e-8)) {}
+
+  void update(const vec_t &dW, vec_t &W, bool parallelize) {
+    vec_t &mt = get<0>(W);
+    vec_t &vt = get<1>(W);
+
+    for_i(parallelize, static_cast<int>(W.size()), [&](int i) {
+      mt[i] = b1 * mt[i] + (float_t(1) - b1) * dW[i];
+      vt[i] = std::max(b2 * vt[i], std::abs(dW[i]));
+
+      W[i] -= alpha * (mt[i] / (float_t(1) - b1_t)) / (vt[i] + eps);
```

Comment:
not sure what trick you applied here, I propose
`W[i] -= alpha * mt[i] / ((1.0 - b1_t) * vt[i] + eps)`
or just
`W[i] -= (alpha / (1.0 - b1_t)) * (mt[i] / (vt[i] + eps))`